examples/generate_image_dataset/generate_poses.py

```diff
@@ -38,24 +41,24 @@
     df_columns = ["scene_id", "location_x", "location_y", "location_z", "rotation_pitch", "rotation_yaw", "rotation_roll"]
     df = pd.DataFrame(columns=df_columns)
 
+    # configure system based on config
+    spear.configure_system(config)
+
+    # create spear.Instance object
+    sp_instance = spear.Instance(config)
+    
     # iterate over all scenes
     for scene_id in scene_ids:
 
         spear.log("Processing scene: " + scene_id)
 
-        # update scene_id
-        config.defrost()
-        config.SIMULATION_CONTROLLER.SCENE_ID = scene_id
-        config.freeze()
-
-        # create Env object
-        env = spear.Env(config)
-
-        # reset the simulation
-        _ = env.reset()
+        spear.open_level(sp_instance, scene_id)
 
         # get a few random points
-        points = env.get_random_points(args.num_poses_per_scene)
+        spear.begin_tick(sp_instance)
```

Comment:
```
        instance.engine_service.begin_tick()
        points = instance.legacy_service.get_random_points(args.num_poses_per_scene)
        instance.engine_service.tick()
        instance.engine_service.end_tick()
```

Reply:
We also need to unpause the game after `begin_tick()`, and pause the game towards `end_tick()`. With just using `engine_service.begin_tick()`, we won't be able to achieve that. We need to bring in `game_world_service.pause/unpause` here.

Reply:
Maybe adding these composite function calls to `Env`' as static methods might be better. But, still doesn't seem to be the right place for it.

Reply:
- Are you sure we need to unpause the game to sample from a nav mesh? If so, it is fine if we also pause and unpause the game here, but I would be surprised if we needed to do that.
- The `Env` class is a less fundamental abstraction now. It used to be the only way to interact with the C++ code, but now it is just an optional convenience wrapper on top of the services. Because the `Env` class is less fundamental in our current design, it is fine with me if each example app defines its own. We are already doing this with our `CustomEnv` and `OpenBotEnv` classes. I'm just suggesting to take this idea a step further and remove the `Env` abstraction from the `spear` module entirely. If we find that we're repeating the same function in every custom `Env` class, we can define a common `Env` base class in `examples/common/env.py`. If an example doesn't really need an `Env` class because it is too simple, then the example could just interact with the underlying services directly. If we follow this approach, it would not be weird if the `Env` class for this specific example defined its own `get_random_points` function. Note that we do not need to make each `Env` classes conform to the OpenAI Gym interface. Each `Env` class could just define the minimal set of helper functions to support a specific example app.
- It is reasonable to expect that higher-level blocks of code, e.g., `generate_poses.py` or an `Env` class, will need to accomplish their high-level goals by calling multiple services. At a minimum, they'll typically need `engine_service` for the tick functions and `game_world_service` to do anything else. This is a feature, not a bug. The fact that our fundamental abstractions (the services) do not attempt to compose functionality enables them to provide general-purpose control over the Unreal Engine. If it takes 10 function calls into the Unreal Engine to accomplish some higher-level goal, I expect it to also take roughly 10 function calls into the `spear` module to accomplish the same goal. This is a good thing because it means that we're giving users the full expressive power of the Unreal Engine.
- I agree that it is inconvenient for users to clutter their code with low-level service calls everywhere. But the correct solution to that problem is for the user to write their own abstractions that compose higher-level functionality. It would be bad if we attempted to build this higher-level functionality directly into the `spear` module, because we don't have the right amount of context at that level to understand what the user is trying to do.
- A useful analogy here is OpenGL. You wouldn't want to clutter your application code with low-level OpenGL calls everywhere. But that doesn't imply doesn't mean that OpenGL should be written in a higher-level way with your specific application in mind, i.e., it would be a bad design for OpenGL to have a high-level animation system built into it. OpenGL is fine how it is, but a user should write abstractions on top of raw OpenGL that make sense in their application. Likewise, a `spear` user should write abstractions on top of raw `spear` calls that make sense in their application.
- I'm fine with providing examples following this approach, which is why I'm suggesting for each example to define its own `Env` class if necessary.